### PR TITLE
feat: modify computetask failure report

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Companion PR

- https://github.com/Substra/substra-backend/pull/727
- https://github.com/Substra/substra-frontend/pull/240 

## Description
Modify `FailureReport`:
- add field `asset_type` containing the kind of asset the failure report connect to
- rename `compute_task_key` to `asset_key`, which is a [wire compatible change](https://groups.google.com/g/protobuf/c/hX4Mj0P4N0w) (i.e. does not need to be declared as a new field)

## How has this been tested?

 As this is going to be merged on a branch that is going to be merged to a POC branch, we use MNIST as a baseline of a working model. We will deal with failing tests on the POC before merging on main.

The e2e tests are also broken due to an issue on producing dumps during release, but passed locally.

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

File: e2e/client/option.go [hunk 1]
```diff
@@ -353,7 +353,6 @@ func DefaultSimpleFunctionOptions() *FunctionOptions {
 		Outputs: map[string]*asset.FunctionOutput{
 			"model": {Kind: asset.AssetKind_ASSET_MODEL},
 		},
-		Status: asset.FunctionStatus_FUNCTION_STATUS_CREATED,
```
Comment (SdgJlbl) [inline e2e/client/option.go:356]: Is this line removed because the default status is UNKNOWN now? or is it unrelated?

File: e2e/client/client.go [hunk 2]
```diff
@@ -628,12 +629,12 @@ func (c *TestClient) RegisterFailureReport(taskRef string) *asset.FailureReport
...
 
-func (c *TestClient) GetFailureReport(taskRef string) *asset.FailureReport {
+func (c *TestClient) GetFailureReport(assetRef string) *asset.FailureReport {
 	param := &asset.GetFailureReportParam{
-		ComputeTaskKey: c.ks.GetKey(taskRef),
+		AssetKey: c.ks.GetKey(assetRef),
 	}
 
-	c.logger.Debug().Str("task key", param.ComputeTaskKey).Msg("getting failure report")
+	c.logger.Debug().Str("asset key", param.AssetKey).Msg("getting failure report")
```
Comment (SdgJlbl) [inline e2e/client/client.go:637]: Do we also want to log the AssetType?

File: e2e/computetask_test.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ package e2e
 
 import (
 	"fmt"
-	"sync"
+	// "sync"
```
Comment (SdgJlbl) [inline e2e/computetask_test.go:8]: 🕵️

File: lib/asset/failure_report_validation_test.go [hunk 4]
```diff
@@ -19,40 +19,47 @@ func TestFailureReportValidate(t *testing.T) {
 
 	cases := map[string]failureReportTestCase{
 		"empty": {&NewFailureReport{}, false},
-		"invalidComputeTaskKey": {&NewFailureReport{
-			ComputeTaskKey: "notUUID",
-			ErrorType:      ErrorType_ERROR_TYPE_BUILD,
-			LogsAddress:    nil,
+		"invalidAssetKey": {&NewFailureReport{
```
Comment (SdgJlbl) [inline lib/asset/failure_report_validation_test.go:22]: Are we okay with not covering all combinations in this test? (like `FailedAssetKind_FAILED_ASSET_COMPUTE_TASK + "notUUID"`)

File: lib/asset/sql_test.go [hunk 5]
```diff
@@ -90,3 +90,17 @@ func TestEventKindValue(t *testing.T) {
 
 	assert.Equal(t, kind, scanned)
 }
+
+func TestFailedAssetKindKindValue(t *testing.T) {
+	k := FailedAssetKind_FAILED_ASSET_UNKNOWN
+	kind := &k
```
Comment (SdgJlbl) [inline lib/asset/sql_test.go:96]: I'm curious if there is a specific reason to write this in 2 lines?

File: lib/service/failure_report_test.go [hunk 6]
```diff
@@ -12,6 +12,7 @@ import (
 	orcerrors "github.com/substra/orchestrator/lib/errors"
 )
 
+// TODO: add tests with function
```
Comment (SdgJlbl) [inline lib/service/failure_report_test.go:15]: Linear card?

**Review by SdgJlbl — APPROVED** [review]

File: e2e/client/option.go (hunk 1 above)
Comment (guilhem-barthes) [reply e2e/client/option.go:356]: It's a mistake I made visibly in the other PR, but I was too lazy to rebase again, but I can do it if it's disturbing you! I thought this struct was a `Function` but it is a `FunctionOption`, which doesn't have a `Status` field

File: e2e/client/client.go (hunk 2 above)
Comment (guilhem-barthes) [reply e2e/client/client.go:637]: good point !

File: e2e/computetask_test.go [hunk 7]
```diff
@@ -227,41 +227,41 @@ func TestQueryTasks(t *testing.T) {
...
-			client2.RegisterTasks(client.DefaultTrainTaskOptions().
-				WithKeyRef(fmt.Sprintf("task2%d", i)).
-				WithInput("model", &client.TaskOutputRef{TaskRef: "parent2", Identifier: "model"}))
-			wg.Done()
-		}(i)
-	}
-
-	wg.Wait()
-}
+// func TestConcurrency(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (guilhem-barthes) [reply e2e/computetask_test.go:230]: No,, thanks for mentioning it, this test is failing locally (but pass on the CI) and I visibly committed commented :x

File: lib/asset/sql.go [hunk 8]
```diff
@@ -155,3 +155,26 @@ func (fs *FunctionStatus) Scan(value interface{}) error {
 
 	return nil
 }
+
+// Value implements the driver.Valuer interface.
+// Simply returns the string representation of the FunctionStatus.
+func (fak *FailedAssetKind) Value() (driver.Value, error) {
+	return fak.String(), nil
+}
```
Comment (oleobal) [inline lib/asset/sql.go:163]: Can this function ever return an error?

File: server/standalone/dbal/failure_report_test.go [hunk 9]
```diff
@@ -18,17 +17,17 @@ func TestGetFailureReportNotFound(t *testing.T) {
 
 	mock.ExpectBegin()
 
-	computeTaskKey := "4c67ad88-309a-48b4-8bc4-c2e2c1a87a83"
+	assetKey := "4c67ad88-309a-48b4-8bc4-c2e2c1a87a83"
 	mock.ExpectQuery(`SELECT .* FROM expanded_failure_reports`).
-		WithArgs(testChannel, computeTaskKey).
+		WithArgs(assetKey, testChannel).
```
Comment (oleobal) [inline server/standalone/dbal/failure_report_test.go:22]: Is reversing the order of the arguments of any importance?

**Pull request merged**